compiler/rustc_macros/src/diagnostics/utils.rs

```diff
@@ -477,7 +477,12 @@ impl SubdiagnosticKind {
     pub(super) fn from_attr(
         attr: &Attribute,
         fields: &impl HasFieldMap,
-    ) -> Result<(SubdiagnosticKind, Option<Path>), DiagnosticDeriveError> {
+    ) -> Result<(Option<SubdiagnosticKind>, Option<Path>), DiagnosticDeriveError> {
```

Comment:
```suggestion
    ) -> Result<Option<(SubdiagnosticKind, Option<Path>)>, DiagnosticDeriveError> {
```

`Ok((None, Some(path)))` wouldn't make sense.

Reply:
Fixed!